packages/auto-label/src/auto-label.ts

```diff
@@ -314,15 +334,67 @@ export function handler(app: Application) {
     );
   });
 
+  app.on(['pull_request.opened'], async context => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const config: any = await context.config('config.yml', DEFAULT_CONFIGS);
+    if (!config.language) return;
+    if (!config.language.pullrequest) return;
+    if (langlabler.langLabelExists(context)) return;
+    logger.info(
+      'Labeling New Pull Request: ' +
+        context.payload.repository.name +
+        ' #' +
+        context.payload.pull_request.number
+    );
+    const owner = context.payload.repository.owner.login;
+    const repo = context.payload.repository.name;
+    const pull_number = context.payload.pull_request.number;
+    const filesChanged = await context.github.pulls.listFiles({
+      owner,
+      repo,
+      pull_number,
+    });
+    const language = langlabler.getPRLanguage(
+      filesChanged.data,
+      config.language
+    );
+    if (language) {
+      logger.info('Labeling PR with: ' + language);
+      await context.github.issues.addLabels({
+        owner,
+        repo,
+        issue_number: pull_number,
+        labels: [language],
+      });
+    }
+  });
+
   app.on(['installation.created'], async context => {
     const repositories = context.payload.repositories;
     const driftRepos = await handler.getDriftRepos();
-    if (!driftRepos) {
-      return;
-    }
+    if (!LABEL_PRODUCT_BY_DEFAULT) return;
+    if (!driftRepos) return;
+
     for await (const repository of repositories) {
       const [owner, repo] = repository.full_name.split('/');
 
+      // Looks for a config file, breaks if user disabled product labels
+      const response = await context.github.repos.getContent({
+        owner,
+        repo,
+        path: '.github/config.yml',
```

Comment:
let's make this something specific to auto-label, I gave the bot permission to access `.github/autolabel.yaml`